adapters/oidc/tomcat/tomcat-core/src/main/java/org/keycloak/adapters/tomcat/AbstractKeycloakAuthenticatorValve.java

```diff
@@ -68,7 +68,7 @@ public void lifecycleEvent(LifecycleEvent event) {
             cache = false;
         } else if (Lifecycle.AFTER_START_EVENT.equals(event.getType())) {
         	keycloakInit();
-        } else if (event.getType() == Lifecycle.BEFORE_STOP_EVENT) {
+        } else if (event.getType().equals(Lifecycle.BEFORE_STOP_EVENT)) {
```

Comment:
Please swap the two sides of the `equals` operation so that they are similar to the statement above. The new order has the advantage that the first part can never be null, so I consider it a best practice.